probe-rs/src/flashing/download.rs

```diff
@@ -40,39 +46,67 @@ pub enum FileDownloadError {
     Object(&'static str),
 }
 
-/// Downloads a file at `path` into flash.
+/// Downloads a file of given `format` at `path` to the flash with progress reporting.
+///
+/// This will ensure that memory bounderies are honored and does unlocking, erasing and programming of the flash for you.
+///
+/// If `keep_unwritten_bytes` is `true`, erased portions that are not overwritten by the ELF data
+/// are restored afterwards, such that the old contents are untouched.
+///
+/// If no progress reporting is desired, have a look at `flashing::download_file()`.
 pub fn download_file_with_progress_reporting(
     session: &Session,
     path: &Path,
     format: Format,
     memory_map: &[MemoryRegion],
```

Comment:
Do we need the memory_map as a separate parameter? I think it's contained in the session anyway.

Reply:
Fixed.